lib/puppet/provider/rh_subscription/redhat.rb

```diff
@@ -1,41 +1,121 @@
+  #             OU: MyOU
+  #             ST: MyST
+  #
+  # Product:
+  #             ID: 69
+  #             Name: Red Hat Enterprise Linux Server
+  #             Version:
+  #             Arch: x86_64,ia64,x86
+  #             Tags:
+  #             Brand Type: OS
+  #             Brand Name: Red Hat Enterprise Linux
+  #
+  # Order:
+  #             Name: Red Hat Enterprise Linux Server with Smart Management
+  #             Number: 12345678
+  #             SKU: RH00009
+  #
+  #
+  # In order to easily access the information provided by "rct cat-cert",
+  # the output is parsed into a structure like:
+  #
+  #   {
+  #     Certificate: {
+  #       "Path": "/etc/pki/entitlement/123421432423233332.pem",
+  #       ...
+  #     },
+  #     ...
+  #     "Order": {
+  #       "Name": "Red Hat Enterprise Linux Server with Smart Management",
+  #       "Number": "12345678",
+  #       "SKU": "RH00009"
+  #     }
+  #   }
+  #
+  # @returns a hash containing all active subscriptions
+  #   Key: pool-id
+  #   Value: hash containing the name of the subscription (Order/Name) and
+  #     the subscription's certificate serial (Certificate/Serial).
+  #
+  #   Example:
+  #     {
+  #       "1fe7cf5f675fbca60167cadd311232118": {
+  #         name: "Red Hat Enterprise Linux Server with Smart Management",
+  #         serial: "123421432423233332"
+  #       },
+  #       "2f3278127287f2900919eabc111123122": {
+  #         name: "Custom-Product",
+  #         serial: "375782838583892982"
+  #       }
+  #     }
+  #
   def self.subscriptions
     subs = {}
-    rhsm('list', '--consumed').split("\n\n").each do |blk|
+    Dir.glob('/etc/pki/entitlement/*.pem') do |cert|
+      next if cert =~ %r{-key.pem$}
+      cert_raw_data = rct('cat-cert', cert)
       sub_data = {}
-      blk.split("\n").each do |line|
-        element = line.match(%r{^([^:]+):[^\w]+(\w.*)})
-        next unless element.is_a?(MatchData)
-        sub_data[element[1]] = element[2]
+      section = nil
+      cert_raw_data.split("\n").each do |line|
+        if line =~ %r{^([^:\s]+):$}
+          section = Regexp.last_match(1).strip
+          sub_data[section] = {}
+        end
+        next if section.nil?
+        sub_data[section][Regexp.last_match(1).strip] = Regexp.last_match(2).strip if line =~ %r{^\s+([^:]+):(.+)$}
       end
-      subs[sub_data['Pool ID']] = {
-        name: sub_data['Subscription Name'],
-        active: sub_data['Active'],
-        serial: sub_data['Serial']
+      subs[sub_data['Certificate']['Pool ID']] = {
+        name: sub_data['Order']['Name'],
+        serial: sub_data['Certificate']['Serial']
       }
     end
     subs
   end
 
-  def self.active?(str)
-    case str
-    when 'True'
-      :true
-    when 'False'
-      :false
-    end
-  end
-
   def self.instances
     subscriptions.map do |pool, data|
       new(
         ensure: :present,
         name:   pool,
-        active: active?(data[:active]),
         serial: data[:serial]
```

Comment:
Rubocop mentions the missing comma at the end of the line

Reply:
If I add the comma, TravisCI build fails...

Offenses:

lib/puppet/provider/rh_subscription/redhat.rb:108:50: C: Style/TrailingCommaInLiteral: Avoid comma after the last item of a hash. (https://github.com/bbatsov/ruby-style-guide#no-trailing-array-commas)

        serial: sub_data['Certificate']['Serial'],

                                                 ^

lib/puppet/provider/rh_subscription/redhat.rb:119:30: C: Style/TrailingCommaInArguments: Avoid comma after the last parameter of a method call. (https://github.com/bbatsov/ruby-style-guide#no-trailing-params-comma)

        serial: data[:serial],

Reply:
Yes, in Rubocop we have different rules. Personally I'd prefer trailing commas here as well but let's not change rubocop rules as part of this PR.